vault/core.go

```diff
@@ -1244,9 +1270,20 @@ func (c *Core) sealInternal() error {
 		return err
 	}
 	c.logger.Printf("[INFO] core: vault is sealed")
+
 	return nil
 }
 
+// stepDownInternal is an internal method used to step down from active duty.
+// It does not do any authorization checking.
+func (c *Core) stepDownInternal() {
+	// Merely trigger the loop to re-run. This value will cause the
+	// loop to run through giving up leadership, but without triggering
+	// the return at the end of the next loop run, since it's not
+	// closed
+	c.standbyStopCh <- struct{}{}
```

Comment:
The problem is `runStandby` will check the top of the loop for *any* value and exit the loop, so depending on timing this can cause the standby mode to suddenly break. I would instead change the channel to take a bool, where the default `false` value indicates the loop should break, while `true` indicates a user initiated step down. We can then modify the logic in `runStandby` to not return on the `true` value in a way that is race free.